tests/System.Text.Json.Tests/JsonObjectTests.cs

```diff
@@ -20,6 +20,7 @@ public void ParseArray()
 
                 Assert.Equal(phoneNumber, "425-214-3151");
                 Assert.Equal(age, 25);
+                Assert.Equal(parsedObject.Length, 2);
```

Comment:
I would move this check before `var phoneNumber = (string)parsedObject[0];` because if this fails, we don't want to get out of range exception.